checker/jdk/nullness/src/javax/script/ScriptEngineManager.java

```diff
@@ -0,0 +1,421 @@
+                        return engine;
+                    } catch (Exception exp) {
+                        if (DEBUG) exp.printStackTrace();
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns a list whose elements are instances of all the <code>ScriptEngineFactory</code> classes
+     * found by the discovery mechanism.
+     * @return List of all discovered <code>ScriptEngineFactory</code>s.
+     */
+    public List<ScriptEngineFactory> getEngineFactories() {
+        List<ScriptEngineFactory> res = new ArrayList<ScriptEngineFactory>(engineSpis.size());
+        for (ScriptEngineFactory spi : engineSpis) {
+            res.add(spi);
+        }
+        return Collections.unmodifiableList(res);
+    }
+
+    /**
+     * Registers a <code>ScriptEngineFactory</code> to handle a language
+     * name.  Overrides any such association found using the Discovery mechanism.
+     * @param name The name to be associated with the <code>ScriptEngineFactory</code>.
+     * @param factory The class to associate with the given name.
+     * @throws NullPointerException if any of the parameters is null.
+     */
+    public void registerEngineName(String name, ScriptEngineFactory factory) {
+        if (name == null || factory == null) throw new NullPointerException();
+        nameAssociations.put(name, factory);
+    }
+
+    /**
+     * Registers a <code>ScriptEngineFactory</code> to handle a mime type.
+     * Overrides any such association found using the Discovery mechanism.
+     *
+     * @param type The mime type  to be associated with the
+     * <code>ScriptEngineFactory</code>.
+     *
+     * @param factory The class to associate with the given mime type.
+     * @throws NullPointerException if any of the parameters is null.
+     */
+    public void registerEngineMimeType(String type, ScriptEngineFactory factory) {
+        if (type == null || factory == null) throw new NullPointerException();
+        mimeTypeAssociations.put(type, factory);
+    }
+
+    /**
+     * Registers a <code>ScriptEngineFactory</code> to handle an extension.
+     * Overrides any such association found using the Discovery mechanism.
+     *
+     * @param extension The extension type  to be associated with the
+     * <code>ScriptEngineFactory</code>.
+     * @param factory The class to associate with the given extension.
+     * @throws NullPointerException if any of the parameters is null.
+     */
+    public void registerEngineExtension(String extension, ScriptEngineFactory factory) {
+        if (extension == null || factory == null) throw new NullPointerException();
+        extensionAssociations.put(extension, factory);
+    }
+
+    /** Set of script engine factories discovered. */
+    private @SuppressWarnings("initialization") HashSet<ScriptEngineFactory> engineSpis;    // gets initialized in init method which is called by constuctor of class
```

Comment:
In this case, the `init()` method should be annotated as setting the variables.  No annotations are needed on the fields.

Also, `@SuppressWarnings` should be on the line before `private`, not after `private`.